### PR TITLE
Add Pyfa

### PR DESCRIPTION
**Pull request by asbytes**
### Please confirm your submission meets all the criteria

- [x] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [x] My pull request follows the instructions at [App Submission][submission].
- [x] I am using only the minimal set of permissions. *(If not, please explain each non-standard permission.)*
- [x] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [x] I am an upstream contributor to the project. If not, I contacted upstream developers about submitting their software to Flathub. **https://github.com/pyfa-org/Pyfa/issues/2510#issuecomment-1590195003**
- [x] I own the domain used in the application ID or the domain has a policy for delegating subdomains (e.g. GitHub, SourceForge).
- [ ] Any additional patches or files have been submitted to the upstream projects concerned. *All included patches are related to GTK2 and WebKitGTK and were taken from GIMP flatpak*

[reqs]: https://docs.flathub.org/docs/for-app-authors/requirements
[maint]: https://docs.flathub.org/docs/for-app-authors/maintanance
[submission]: https://docs.flathub.org/docs/for-app-authors/submission

### CODE REVIEW

File: com.github.pyfa_org.Pyfa.json [hunk 1]
```diff
@@ -0,0 +1,347 @@
+{
+  "id" : "com.github.pyfa_org.Pyfa",
```
Comment (hfiguiere) [inline com.github.pyfa_org.Pyfa.json:2]: it should be `io.github`

File: com.github.pyfa_org.Pyfa.json [hunk 2]
```diff
@@ -0,0 +1,347 @@
...
+  "branch" : "stable",
```
Comment (hfiguiere) [inline com.github.pyfa_org.Pyfa.json:3]:
```suggestion
```

File: com.github.pyfa_org.Pyfa.json [hunk 3]
```diff
@@ -0,0 +1,347 @@
...
+  "runtime" : "org.gnome.Platform",
+  "runtime-version" : "43",
```
Comment (hfiguiere) [inline com.github.pyfa_org.Pyfa.json:5]:
```suggestion
  "runtime-version" : "44",
```

File: com.github.pyfa_org.Pyfa.json [hunk 4]
```diff
@@ -0,0 +1,347 @@
...
+  "command" : "pyfa",
+  "finish-args" : [
+    "--share=ipc",
+    "--share=network",
+    "--socket=x11",
+    "--filesystem=xdg-config/gtk-3.0"
+  ],
+  "modules" : [
+    {
+      "name" : "python-3.7",
```
Comment (hfiguiere) [inline com.github.pyfa_org.Pyfa.json:16]: why do you need to build python? It is in the runtime.

File: com.github.pyfa_org.Pyfa.json [hunk 5]
```diff
@@ -0,0 +1,347 @@
...
+          {
+              "type": "script",
+              "commands": [
+                  "autoreconf -fsi"
+              ]
+          }
+      ],
+      "modules": [
+          {
+              "name": "gtk2",
```
Comment (hfiguiere) [inline com.github.pyfa_org.Pyfa.json:102]: please use the one from shared-modules

https://github.com/flathub/shared-modules/

File: flathub.json [hunk 6]
```diff
@@ -0,0 +1,3 @@
+{
+  "only-arches": ["x86_64"]
```
Comment (hfiguiere) [inline flathub.json:2]: this is disappointing.

File: com.github.pyfa_org.Pyfa.metainfo.xml [hunk 7]
```diff
@@ -0,0 +1,44 @@
...
+    </screenshot>
+  </screenshots>
+  <releases>
+    <release version="2.53.0" date="2023-06-13"/>
+    <release version="2.52.0" date="2023-05-25"/>
+    <release version="2.51.0" date="2023-03-16"/>
+    <release version="2.50.0" date="2023-02-18"/>
+    <release version="2.49.0" date="2022-12-21"/>
+    <release version="2.48.0" date="2022-11-10"/>
+  </releases>
```
Comment (hfiguiere) [inline com.github.pyfa_org.Pyfa.metainfo.xml:43]: missing required `content_rating`

File: com.github.pyfa_org.Pyfa.json (hunk 3 above)
Comment (hfiguiere) [reply com.github.pyfa_org.Pyfa.json:5]: also you build gtk2, so why this runtime?

File: com.github.pyfa_org.Pyfa.json [hunk 8]
```diff
@@ -0,0 +1,347 @@
...
+        "install -D --mode=0755 pyfa /app/bin/pyfa",
+        "install -D -t \"${FLATPAK_DEST}/share/applications/\" ./com.github.pyfa_org.Pyfa.desktop",
+        "install -D ./com.github.pyfa_org.Pyfa-256.png \"${FLATPAK_DEST}/share/icons/hicolor/256x256/apps/com.github.pyfa_org.Pyfa.png\"",
+        "install -D -t \"${FLATPAK_DEST}/share/metainfo/\" ./com.github.pyfa_org.Pyfa.metainfo.xml"
+      ]
+    }
+  ],
+  "cleanup" : [
+    "/lib/cmake/",
+    "/lib/debug",
```
Comment (hfiguiere) [inline com.github.pyfa_org.Pyfa.json:287]:
```suggestion
```

File: com.github.pyfa_org.Pyfa.json [hunk 9]
```diff
@@ -0,0 +1,347 @@
...
+        "cp -r eos graphs gui imgs locale service utils eve.db config.py pyfa.py db_update.py README.md LICENSE version.yml ${FLATPAK_DEST}/lib/pyfa",
+        "find ${FLATPAK_DEST}/lib/pyfa | grep -E '(__pycache__|\\.pyc|\\.pyo$)' | xargs rm -rf",
+        "install -D --mode=0755 pyfa /app/bin/pyfa",
+        "install -D -t \"${FLATPAK_DEST}/share/applications/\" ./com.github.pyfa_org.Pyfa.desktop",
+        "install -D ./com.github.pyfa_org.Pyfa-256.png \"${FLATPAK_DEST}/share/icons/hicolor/256x256/apps/com.github.pyfa_org.Pyfa.png\"",
+        "install -D -t \"${FLATPAK_DEST}/share/metainfo/\" ./com.github.pyfa_org.Pyfa.metainfo.xml"
+      ]
+    }
+  ],
+  "cleanup" : [
```
Comment (hfiguiere) [inline com.github.pyfa_org.Pyfa.json:285]: cleanup should be above modules.

Also it contain a long list of things that don't even seem to be built so maybe it should be trimmed to what's actually needed.

File: com.github.pyfa_org.Pyfa.json [hunk 10]
```diff
@@ -0,0 +1,347 @@
...
+    "/bin/pyslices",
+    "/bin/pyslicesshell",
+    "/bin/pywxrc",
+    "/bin/roman",
+    "/bin/wheel",
+    "/bin/wxdemo",
+    "/bin/wxdocs",
+    "/bin/wxget"
+  ],
+  "separate-locales" : false,
```
Comment (hfiguiere) [inline com.github.pyfa_org.Pyfa.json:343]: shuld be above modules

File: com.github.pyfa_org.Pyfa.json [hunk 11]
```diff
@@ -0,0 +1,347 @@
...
+  "tags" : [
+    "stable"
+  ]
```
Comment (hfiguiere) [inline com.github.pyfa_org.Pyfa.json:346]:
```suggestion
```

you don't need this.

File: com.github.pyfa_org.Pyfa.metainfo.xml [hunk 12]
```diff
@@ -0,0 +1,44 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<component type="desktop">
+  <id>com.github.pyfa_org.Pyfa</id>
+  <launchable type="desktop-id">com.github.pyfa_org.Pyfa.desktop</launchable>
+  <name>Pyfa</name>
+  <developer_name>pyfa-org</developer_name>
+  <summary>Python fitting assistant for EVE: Online</summary>
+  <metadata_license>MIT</metadata_license>
+  <project_license>GPL-3</project_license>
```
Comment (hfiguiere) [inline com.github.pyfa_org.Pyfa.metainfo.xml:9]: this is an invalid SPDX license identifier.

File: com.github.pyfa_org.Pyfa.json [hunk 13]
```diff
@@ -0,0 +1,347 @@
...
+          },
+          {
+              "type": "patch",
+              "path": "patches/webkitgtk-41-fix.patch"
+          }
+      ]
+    },
+    "python3-modules.json",
+    {
+      "name": "pyfa-2.53.0",
```
Comment (hfiguiere) [inline com.github.pyfa_org.Pyfa.json:246]:
```suggestion
      "name": "pyfa",
```

File: com.github.pyfa_org.Pyfa.json [hunk 14]
```diff
@@ -0,0 +1,347 @@
...
+  "branch" : "stable",
+  "runtime" : "org.gnome.Platform",
+  "runtime-version" : "43",
+  "sdk" : "org.gnome.Sdk",
+  "command" : "pyfa",
+  "finish-args" : [
+    "--share=ipc",
+    "--share=network",
+    "--socket=x11",
+    "--filesystem=xdg-config/gtk-3.0"
```
Comment (hfiguiere) [inline com.github.pyfa_org.Pyfa.json:12]: why do you need this ?

File: com.github.pyfa_org.Pyfa.json [hunk 15]
```diff
@@ -0,0 +1,347 @@
...
+      "build-commands": [
+        "find locale/ -type f -name '*.po' -exec msgen '{}' -o '{}' \\;",
+        "/app/bin/python3.7 scripts/compile_lang.py",
+        "/app/bin/python3.7 db_update.py",
+        "mkdir -p ${FLATPAK_DEST}/lib/pyfa",
+        "cp -r eos graphs gui imgs locale service utils eve.db config.py pyfa.py db_update.py README.md LICENSE version.yml ${FLATPAK_DEST}/lib/pyfa",
+        "find ${FLATPAK_DEST}/lib/pyfa | grep -E '(__pycache__|\\.pyc|\\.pyo$)' | xargs rm -rf",
+        "install -D --mode=0755 pyfa /app/bin/pyfa",
+        "install -D -t \"${FLATPAK_DEST}/share/applications/\" ./com.github.pyfa_org.Pyfa.desktop",
+        "install -D ./com.github.pyfa_org.Pyfa-256.png \"${FLATPAK_DEST}/share/icons/hicolor/256x256/apps/com.github.pyfa_org.Pyfa.png\"",
```
Comment (hfiguiere) [inline com.github.pyfa_org.Pyfa.json:280]:
```suggestion
        "install -D ./com.github.pyfa_org.Pyfa-256.png ${FLATPAK_DEST}/share/icons/hicolor/256x256/apps/com.github.pyfa_org.Pyfa.png",
```

File: com.github.pyfa_org.Pyfa.json [hunk 16]
```diff
@@ -0,0 +1,347 @@
...
+        "install -D -t \"${FLATPAK_DEST}/share/metainfo/\" ./com.github.pyfa_org.Pyfa.metainfo.xml"
```
Comment (hfiguiere) [inline com.github.pyfa_org.Pyfa.json:281]:
```suggestion
        "install -D -t ${FLATPAK_DEST}/share/metainfo/ ./com.github.pyfa_org.Pyfa.metainfo.xml"
```

File: com.github.pyfa_org.Pyfa.json [hunk 17]
```diff
@@ -0,0 +1,347 @@
...
+      "buildsystem": "simple",
+      "build-commands": [
+        "find locale/ -type f -name '*.po' -exec msgen '{}' -o '{}' \\;",
+        "/app/bin/python3.7 scripts/compile_lang.py",
+        "/app/bin/python3.7 db_update.py",
+        "mkdir -p ${FLATPAK_DEST}/lib/pyfa",
+        "cp -r eos graphs gui imgs locale service utils eve.db config.py pyfa.py db_update.py README.md LICENSE version.yml ${FLATPAK_DEST}/lib/pyfa",
+        "find ${FLATPAK_DEST}/lib/pyfa | grep -E '(__pycache__|\\.pyc|\\.pyo$)' | xargs rm -rf",
+        "install -D --mode=0755 pyfa /app/bin/pyfa",
+        "install -D -t \"${FLATPAK_DEST}/share/applications/\" ./com.github.pyfa_org.Pyfa.desktop",
```
Comment (hfiguiere) [inline com.github.pyfa_org.Pyfa.json:279]:
```suggestion
        "install -D -t ${FLATPAK_DEST}/share/applications/ ./com.github.pyfa_org.Pyfa.desktop",
```

File: com.github.pyfa_org.Pyfa.json (hunk 4 above)
Comment (asbytes) [reply com.github.pyfa_org.Pyfa.json:15]: Pyfa's official releases are built with python-3.7. Application has unresolved issues when running on recent versions of Python.

File: com.github.pyfa_org.Pyfa.json (hunk 9 above)
Comment (asbytes) [reply com.github.pyfa_org.Pyfa.json:284]: Maybe this list looks excessive, but i tried to reduce final size of application. All these files are present in build and most of them are installed with python deps.

File: com.github.pyfa_org.Pyfa.json (hunk 3 above)
Comment (asbytes) [reply com.github.pyfa_org.Pyfa.json]: Changed runtime to org.freedesktop.Platform//22.08

File: flathub.json (hunk 6 above)
Comment (asbytes) [reply flathub.json:2]: I can't do anything with this. Pyfa releases are built for x86_64. Other architectures have never been claimed to support.

File: com.github.pyfa_org.Pyfa.metainfo.xml (hunk 7 above)
Comment (asbytes) [reply com.github.pyfa_org.Pyfa.metainfo.xml]: Done

File: com.github.pyfa_org.Pyfa.json (hunk 9 above)
Comment (asbytes) [reply com.github.pyfa_org.Pyfa.json]: Cleanup array moved above modules

File: com.github.pyfa_org.Pyfa.metainfo.xml (hunk 12 above)
Comment (asbytes) [reply com.github.pyfa_org.Pyfa.metainfo.xml]: Fixed. Also changed metadata license to CC0-1.0

File: com.github.pyfa_org.Pyfa.json (hunk 14 above)
Comment (asbytes) [reply com.github.pyfa_org.Pyfa.json]: Removed

File: com.github.pyfa_org.Pyfa.json (hunk 10 above)
Comment (asbytes) [reply com.github.pyfa_org.Pyfa.json]: Done

File: com.github.pyfa_org.Pyfa.json (hunk 1 above)
Comment (asbytes) [reply com.github.pyfa_org.Pyfa.json]: Done

File: com.github.pyfa_org.Pyfa.json (hunk 5 above)
Comment (asbytes) [reply com.github.pyfa_org.Pyfa.json]: Done

**Comment by asbytes** [comment]
bot, build io.github.pyfa_org.Pyfa

**Pull request closed without merging**